langserver/symbol.go

```diff
@@ -488,3 +479,91 @@ var symbolCacheTotal = prometheus.NewCounterVec(prometheus.CounterOpts{
 func init() {
 	prometheus.MustRegister(symbolCacheTotal)
 }
+
+func listPkgs(ctxt *build.Context, dir string) []string {
```

Comment:
This code is not at all `ExpandPattern`. I'll give it a better name and documentation though and mention why it is better than `ExpandPattern` and how it relates.

Yeah I'll avoid tackling other expand pattern uses, but I'll add an inline TODO to hint to future travels about perf improvements.